packages/components/src/components/badge/badge.tsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import classNames from 'classnames';
+
+type TBadgeSize = 'medium' | 'large';
+type TBadgeSpacing = 'tight' | 'loose';
+type TWeight = 'bold' | 'regular';
+type TBackgroundColor = 'blue' | 'orange' | 'red' | 'gray';
```

Comment:
if the `type: 'bordered'` the `background_color` would be redundant, so we can combine these two props together and restrict the developer. In this case we can remove the `is_contained` from the 59-62 lines

Reply:
@mahdiyeh-deriv understand your concern, that's why i decided to use Discriminated Unions here. So when the developer use Badge component and try to add `background_color` for `bordered` type, typescript will complain that there is no such property in Badge type.
<img width="592" alt="Screenshot 2023-05-02 at 09 24 00" src="https://user-images.githubusercontent.com/103181646/235593871-6863428e-3101-4840-9da6-d18a8f9dc807.png">
<img width="1146" alt="Screenshot 2023-05-02 at 09 24 46" src="https://user-images.githubusercontent.com/103181646/235593984-d8b3e412-3dc3-46fd-b414-de381b50d304.png">
If you still think that it's not enough/sufficient solution for you concern, we can apply your suggestion. But i'm not fully understand about how do you want combine `type` and `background` property.

Reply:
It's enough, thanks @george-usynin-deriv